### PR TITLE
[FIX] web, mail: searchbar facet cog overlapping

### PR DESCRIPTION
**Pull request by mano-odoo**
When discuss is not installed, hovering a facet in the searchbar makes the cog overlap the icon/text.

This is due to the `o-bg-inherit` class which is set in the mail module. Therefore if discuss is not installed, the SCSS for the background doesn't exist.

This commit moves the `o-bg-inherit` as a new utility class `bg-inherit`, allowing it to be used even if no apps are installed.

Therefore this commit also renames the occurences of `o-bg-inherit` into `bg-inherit`.

task-3575837

image

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by mano-odoo** [comment]
@Brieuc-brd I renamed the o-bg-inherit into o_bg_inherit (in a separate commit for now in case we want to revert), it's ready for your review 😄

**Review by Brieuc-brd — APPROVED** [review]
LGTM 👍 
@stefanorigano I'd like to know what you think of this change :)

**Review by stefanorigano — CHANGES_REQUESTED** [review]
Thanks @mano-odoo .

If we want to extrapolate `o-bg-inherit` from `mail`and make it a default utility-class, we should generate it using the same system as the other ones, meaning either using Bootstrap API or `bootstrap_review_backend.scss`.

so: `bg-primary`, `bg-white`, ... , `bg-inherit` 

cc @Brieuc-brd

File: addons/web/static/src/scss/bootstrap_review_backend.scss [hunk 1]
```diff
@@ -192,6 +192,10 @@ $-o-theme-text-colors: null;
     @extend %-bg-black;
 }
 
+.bg-inherit {
```
Comment (stefanorigano) [inline addons/web/static/src/scss/bootstrap_review_backend.scss:195]: This is not related to white/black classes (even if you slightly modified the note).
I propose to move it in the " _Odoo custom text/bg colors classes generation_" section, in its own point (n6 then), around L269

**Review by stefanorigano — CHANGES_REQUESTED** [review]

File: addons/web/static/src/scss/bootstrap_review_backend.scss (hunk 1 above)
Comment (mano-odoo) [reply addons/web/static/src/scss/bootstrap_review_backend.scss:195]: Understood, I placed it here in regards to the "bg-transparent" utility which felt similar to me but indeed behaves differently since it extends the bg-black rather than applying a `background-color: transparent`.

**Review by stefanorigano — APPROVED** [review]
Thanks @mano-odoo 👍 

@odoo/rd-discuss , congratulation! We're promoting you custom class to default utility-class 🙌😅

Could you check and eventually r+ ? Thanks!

**Review by alexkuhn — APPROVED** [review]
@robodoo r+
Thank you for promoting `bg-inherit` 😊
